src/redux/movieSlice.ts

```diff
@@ -0,0 +1,36 @@
+import { createSlice } from "@reduxjs/toolkit";
+import { SortingOptionsType } from "../types/sortingOptions.type";
+import { RootState } from "./store";
+
+const initialState = {
+  movieId: 0,
+  sortBy: "",
+  sortOrder: "",
+  filterBy: "",
+};
+
+const movieSlice = createSlice({
+  name: "movie",
+  initialState,
+  reducers: {
+    setSortBy(state, action: { payload: SortingOptionsType }) {
+      state.sortBy = action.payload.sortBy;
+      state.sortOrder = action.payload.sortOrder;
+    },
```

Comment:
one reducer === one action

And the sort order means "asc" (ascending) and "desc" (descending) which you don't allow the user to select.